asmcomp/cmm_invariants.ml

```diff
@@ -163,18 +216,31 @@ let rec check env (expr : Cmm.expression) =
       | Recursive -> env_extended
       | Nonrecursive -> env
     in
-    List.iter (fun (_, _, handler, _) -> check env_handler handler) handlers
+    List.iter (fun (_, args, handler, _) ->
+        let env_handler =
+          List.fold_left (fun env (id, _) ->
+              Env.bind_var env (VP.var id) Immutable)
```

Comment:
You could define `let bind_immut env id = Env.bind_var env (VP.var id) Immutable`, and then use it here (`List.fold_left env_handler (List.map fst args)`), and below in `Ctrywith` and `run`.

Reply:
I thought about it (in fact `bind_var` took the mutability as first argument in my first draft, for this reason), but I'm allergic to `List.map fst args`. Maybe a shared `bind_params` would work, I'll try that.